packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.Props.ts

```diff
@@ -31,6 +31,16 @@ export interface ISearchBoxProps extends React.InputHTMLAttributes<HTMLInputElem
   onSearch?: (newValue: any) => void;
 
   /**
+   * Callback executed when the user clears the search box by either clicking 'X' or hitting escape.
+   */
+  onClearClick?: (ev?: any, onClear?: ((ev?: any) => void) | null) => void;
+
+  /**
+   * Callback executed when the user presses escape in the search box.
+   */
+  onEscape?: (ev?: any, onClear?: ((ev?: any) => void) | null) => void;
```

Comment:
if we want to follow this pattern, we should have a type for that `IEventHandler<TEvent>`

Reply:
yup...I guess this is what I was wanting to shoot for. I'll give it a first pass, see if I can work out the interface